.semaphore/semaphore.yml

```diff
@@ -0,0 +1,892 @@
+            - cache restore codecov
+            - ./codecov
+  - name: Lint and Test Tools
+    run:
+      when: "change_in('/tools')"
+    dependencies:
+      - Setup
+      - Check Generated Protocol Buffer Files
+    task:
+      jobs:
+        - name: Lint Tools
+          commands:
+            - cd tools
+            - cache restore linter
+            - './bin/golangci-lint run --no-config --disable-all --enable gosimple,errcheck,govet,unused,goconst,godot --timeout 3m'
+        - name: Test Tools
+          commands:
+            - cd tools
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+  - name: Users Auth Changes
+    dependencies: []
+    run:
+      when: "change_in('users/policies/auth.proto')"
+    task:
+      jobs:
+        - name: Test Users
+          commands:
+            - cd users
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+        - name: Test Bootstrap
+          commands:
+            - cd bootstrap
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+        - name: Test Certs
+          commands:
+            - cd certs
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+        - name: Test Consumers
+          commands:
+            - cd consumers
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+        - name: Test Things
+          commands:
+            - cd things
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+        - name: Test Twins
+          commands:
+            - cd twins
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+        - name: Test Readers
+          commands:
+            - cd readers
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+      secrets:
+        - name: CODECOV_TOKEN
+  - name: Things Auth Changes
+    dependencies: []
+    run:
+      when: "change_in('things/policies/auth.proto')"
```

Comment:
When changes are in Protobuf files, we need all the tests (i.e. add `things/policies/auth.proto` to a `change_in` list for all adapters.

Reply:
Instead of when: `"change_in('things/policies/auth.proto')"` we need to check for changes in **implementation logic of things** - something like `when: "change_in(['things/policies/postgres', 'things/policies/service.go'])"`. Also, no need to run coverage in this block since we did not change the actual code of adapters, so no coverage changes will be present.